app/src/main/java/com/github/se/cyrcle/ui/profile/EditProfileComponent.kt

```diff
@@ -100,9 +99,11 @@ fun EditProfileComponent(
           cancelButton()
 
           saveButton(
-              User(
-                  UserPublic(user.public.userId, username, profilePictureUrl),
-                  UserDetails(firstName, lastName, user.details!!.email)))
+              // Use .copy to not lose the locas session and avoid re-fetching the user from db.
+              user.copy(
+                  public = user.public.copy(username = username, profilePictureCloudPath = ""),
+                  details = user.details?.copy(firstName = firstName, lastName = lastName),
+                  localSession = LocalSession(profilePictureUri = profilePictureUri)))
```

Comment:
One would argue that argument with the same name as the variable given would not be necessary. Nothing blocking, just that it makes code heavier to read for no reason.

Reply:
I'll remove it !